src/goInstallTools.ts

```diff
@@ -115,9 +115,20 @@ function installTools(goVersion: SemVersion, missing?: string[]) {
 
 	outputChannel.appendLine(''); // Blank line for spacing.
 
+	// http.proxy setting takes precedence over environment variables
+	let httpProxy = vscode.workspace.getConfiguration('http').get('proxy');
+	let env = process.env;
+	if (httpProxy) {
+		env = Object.assign({}, process.env, {
+			http_proxy: httpProxy,
```

Comment:
Sounds good. Only thing pending is the failing tests in travis. Can you debug and see if it fails in your local set up as well?